Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -297,6 +306,9 @@ protected override bool OnBackButtonPressed()
 
 		void InsertPageBefore(Page page, Page before)
 		{
+			if (page == null || before == null)
+				throw new ArgumentException($"Both {nameof(page)} and {nameof(before)} must be initialized.");
```

Comment:
there is no problem in unit testing or checking argument on private api, but it's not required, that's all